[WIP]Expose UI support for custom fields on financial types

pradpnayak opened the pull request:
Overview
----------------------------------------
_A brief description of the pull request. Try to keep it non-technical._

Before
----------------------------------------
_The current status. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

After
----------------------------------------
_What has been changed. Please provide screenshots or gifs ([LICEcap](http://www.cockos.com/licecap/), [SilentCast](https://github.com/colinkeenan/silentcast)) where appropriate._

Technical Details
----------------------------------------
_If the PR introduces noteworthy technical changes, please describe them here. Provide code snippets if necessary_

Comments
----------------------------------------
_Anything else you would like the reviewer to note_

pradpnayak commented:
Can anyone review this for me?

eileenmcnaughton commented:
I haven't reviewed but I peeked at the code & the coding approach taken here is consistent with the work we have been doing in this area.

eileenmcnaughton commented:
@pradpnayak the one issue I have with this is that you are enabling custom fields out of the box. My position so far has been that we should make it possible for extensions to enable these but not do it in core.

My main concern with enabling in core is it sets expectations that we then accept things like making the data available in searches / reports etc & that is a whole extra conversation. I think we should remove that part before we merge it & discuss separately (ok to leave in for now as it might make it easier for a reviewer)

pradpnayak commented on templates/CRM/Financial/Form/FinancialType.tpl:
```diff
@@ -24,46 +24,9 @@
+  <div class="crm-submit-buttons">
+    <a href="{crmURL p='civicrm/admin/financial/financialType/accounts' q="action=browse&reset=1&aid=$aid"}" class="button"><span>{ts}View or Edit Financial Accounts{/ts}</a></span>
+  </div>
+{/if}
```
@eileenmcnaughton I need a help!!

How i can place the div section for button just besides the Save and Cancel buttom? Can i use JS to move the link or will i need to CRM/Core/Form/EntityForm.tpl to include sections insertions?

eileenmcnaughton replied:
This seems ok?

![screenshot 2018-07-24 03 39 32](https://user-images.githubusercontent.com/336308/43087136-3401e1f4-8ef3-11e8-9bb6-fb7270d02536.png)

pradpnayak replied:
When you open in a new tab the 'View or Edit Financial Accounts' button goes below of Save and Cancel

eileenmcnaughton replied:
Does overriding ```protected function addFormButtons()``` { help?

eileenmcnaughton replied:
buttons look ok for me ![screenshot 2018-07-24 03 39 32](https://user-images.githubusercontent.com/336308/43087504-25379e24-8ef4-11e8-9cc8-760f13edb623.png)

eileenmcnaughton replied:
hmm not sure maybe @colemanw has a good idea

I guess we could have an 'inner tpl' of

```
  {if $action eq 8}
    <div class="messages status no-popup">
      <div class="icon inform-icon"></div>
      {$deleteMessage|escape}
    </div>
  {else}
    <table class="form-layout-compressed">
      {foreach from=$entityFields item=fieldSpec}
        {assign var=fieldName value=$fieldSpec.name}
        <tr class="crm-{$entityInClassFormat}-form-block-{$fieldName}">
          {include file="CRM/Core/Form/Field.tpl"}
        </tr>
      {/foreach}
    </table>
    {include file="CRM/common/customDataBlock.tpl"}
  {/if}
```
that is included from CRM/Core/Form/EntityForm.tpl - so you can just include the inner if you want ... different buttons

pradpnayak commented:
@eileenmcnaughton I have updated the PR to include only form changes and moved installation code to an [extension](https://github.com/pradpnayak/org.civicrm.financialtypecustomfield).

eileenmcnaughton commented:
@pradpnayak I tested this in view and edit mode and read through the code. I'm OK to merge this aside from your button alignment issue & I'm OK with that as a follow up too.

eileenmcnaughton commented:
test this please

pradpnayak commented:
Thanks @eileenmcnaughton
